kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/Range.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
+
+import java.util.Objects;
+import java.util.stream.IntStream;
+
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Represents the set of integers between two integer endpoints. So the Range
+ * with startInclusive 1 and endExclusive 3 contains the integers 1 and 2.
+ * A Range must be non-empty, endExclusive must be greater than startInclusive
+ */
+public record Range(int startInclusive, int endExclusive) {
+
+    /**
+     * Constructs a Range
+     * @param startInclusive the (inclusive) initial value
+     * @param endExclusive the exclusive upper bound
+     * @throws IllegalArgumentException if end is before start
+     */
+    public Range {
+        if (endExclusive <= startInclusive) {
+            throw new IllegalArgumentException(
+                    "end of range: " + endExclusive + " (exclusive) is before start of range: " + startInclusive + " (inclusive)");
+        }
```

Comment:
+0.5 
this Range class is modelling the mathematical concept, so I wouldn't overload it with domain logic without renaming it.  if I were going to enforce the 65535 validation (i'm not sure I'd bother), I would do it do it in the caller (RangeAwarePortPerNodeClusterNetworkAddress) rather than here.

Reply:
I see you did just that already.